src/Console/Command/CleanMySQLDatabaseCommand.php

```diff
@@ -60,92 +57,106 @@ protected function configure(): void
-        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+        $pdo              = new PDO($connectionString, $dbUser, $dbPassword);
+        $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
         $databases = array_column($pdo->query('SHOW DATABASES')->fetchAll(), 'Database');
         $databases = array_filter(
             $databases,
-            function ($dbName) use ($pattern) {
+            static function ($dbName) use ($pattern) {
                 return preg_match($pattern, $dbName);
             }
         );
 
         $remove = [];
-        $notfound = [];
 
         foreach ($databases as $database) {
 
             $branchName = !empty($branchNameFilter)
                 ? str_replace($branchNameFilter, '', $database)
                 : $database;
 
+            $issue = null;
             try {
                 $issue = $issueService->get($branchName, ['fields' => ['status']]);
+            } catch (Exception $e) {
+                if ($e->getCode() === 404) {
+                    $logger->critical(
+                        'While cleaning up, I found an unexpected database with no matching Jira Ticket.'
+                        . ' The database will be removed, but you should investigate why it was'
+                        . ' created in the first place!',
+                        [
+                            'CleanMySQLDatabaseName' => 'Error while matching databases to issues',
+                            'Unexpected database'    => $database,
+                            'Branch name'            => $branchName,
+                            'Possible cause'         => 'Jira ticket was deleted'
+                        ]
+                    );
+                } else {
+                    throw $e;
+                }
+            }
 
-                if (!empty($statuses)) {
-                    $issueStatus = strtolower($issue->fields->status->name);
-
-                    if ($invert) {
-                        if (!in_array($issueStatus, $statuses)) {
-                            $remove[] = $database;
-                            $io->text(sprintf('Found orphaned database %s', $database));
-                        }
-                    } else {
-                        if (in_array($issueStatus, $statuses)) {
-                            $remove[] = $database;
-                            $io->text(sprintf('Found orphaned database %s', $database));
-                        }
+            if (!empty($statuses)) {
+                $issueStatus = strtolower($issue->fields->status->name);
+
+                $statusFound = in_array($issueStatus, $statuses, true);
+                if ($invert) {
+                    if (!$statusFound) {
+                        $remove[] = $database;
+                        $io->text(sprintf('Found orphaned database %s', $database));
                     }
+                } elseif ($statusFound) {
```

Comment:
Isn't there a branch missing which covers negation? 

Also in case invert is false we need that to insure the deletion of a db that has no JIRA issue (no issue status present).

Reply:
Fixed by same commit as your other comment